includes/REST_API/Publisher_Logos_Controller.php

```diff
@@ -0,0 +1,428 @@
+				[ 'status' => 400 ]
+			);
+		}
+
+		if ( in_array( $post->ID, $publisher_logos, true ) ) {
+			return new WP_Error(
+				'rest_publisher_logo_exists',
+				__( 'Publisher logo already exists', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		$publisher_logos[] = $post->ID;
+
+		update_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, $publisher_logos );
+
+		$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+
+		if ( 1 === count( $publisher_logos ) || ! in_array( $active_publisher_logo_id, $publisher_logos, true ) ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Removes a publisher logo from the collection.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+		$publisher_logos = array_values( array_diff( $publisher_logos, [ $post->ID ] ) );
+
+		$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+
+		if ( $post->ID === $active_publisher_logo_id || ! in_array( $active_publisher_logo_id, $publisher_logos, true ) ) {
+			// Mark the first available publisher logo as the new default.
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, ! empty( $publisher_logos[0] ) ? $publisher_logos[0] : 0 );
+		}
+
+		update_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, $publisher_logos );
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Updates a publisher logo in the collection.
+	 *
+	 * Can only be used to make it the "active" one.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( $request['active'] ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Get an existing publisher logo's post object, if valid.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param int $id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_publisher_logo( $id ) {
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+
```

Comment:
Should we have a check to see if `$publisher_logos` is an empty array?

Reply:
If it's an empty array the `in_array()` check below will simply fail. So don't see a need for an extra check here, seems redundant.